### DISCUSSION TITLE
Download File_name Change.

### DISCUSSION THREAD

**Discussion by Sandy381** [fbaligand/kibana-enhanced-table#281] [Q&A]
Hi,

I am using the Enhanced table and it is pretty useful in both Kibana & Opensearch!

One issue I'm facing with Enhanced table & Document table is that when i download the Visualization reports, the file_name is 'export.csv'.

I would really love to have 'visualization_name.csv' when i download Raw or Formatted CSV reports.

When I download in data table visualization of kibana, I can download with 'visualization_name.csv'.

Is there any way to change this in settings or build?

Thanks for your response!

**Comment by fbaligand** [comment]
Hi,

If I remember correctly, this issue has been fixed in recent versions?
Could you indicate your kibana version and enhanced table plugin version?

**Reply by Sandy381** [reply]
Hi,

I'm using Opensearch 1.2.0 and enhanced plugin version is enhanced-table-1.13.1_osd-1.2.0.zip.
 When i download it shows export.csv.
It would be helpful if it downloaded as 'visualization_name'.csv

**Reply by fbaligand** [reply]
Well, the issue has been fixed in kibana code itself, since Kibana 7.11.
And open search dashboards is a fork of Kibana 7.10…
So it would require a PR in OSD repository.
Or maybe it has been fixed in last OSD version.

**Reply by fbaligand** [reply]
The change is very simple:
https://github.com/elastic/kibana/pull/84456/files

**Reply by fbaligand** [reply]
I just opened a PR to add missing code so that CSV export filename is "visualization name":
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2192

**Reply by Sandy381** [reply]
Hope this will be pushed in Enhanced table for Opensearch 2.3.0 version.

**Reply by fbaligand** [reply]
I hope so :)

**Reply by fbaligand** [reply]
Good news: the PR has been merged!
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2192

**Reply by fbaligand** [reply]
By the way, a new release has been published and fixed CSV export filename:
https://github.com/fbaligand/kibana-enhanced-table/releases/tag/v1.13.3

So there was an issue both in OSD and in Enhanced Table plugin

**Reply by Sandy381** [reply]
Hi !

I recently checked OSD 2.2.0 enhanced table in my docker.
When i click on Download ( Raw or Formatted) Enhanced table or Document table in any visualization, name of the report is still 'export.csv' and not 'visualization_name.csv'.
The link I used was latest one given: https://github.com/fbaligand/kibana-enhanced-table/releases/download/v1.13.3/enhanced-table-1.13.3_osd-2.2.0.zip
It will be really helpful if we have visualization_name as we are downloading multiple reports and have export.csv creates conflict.
When I click on data table download, I can see'visualization_name.csv' being downloaded.
Any ideas?

**Reply by fbaligand** [reply]
Hi @Sandy381,

As you can see in PR https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2192, it has been labelled "v2.3.0".
So the fix is available only since OSD 2.3.0.
